design/object-bucket/object-bucket.md

```diff
@@ -0,0 +1,148 @@
+# Rook-Ceph Bucket Provisioning
+
+## Proposed Feature
+
+Add an CephObjectBucketClaim API endpoint and control loop to the Rook-Ceph operator for the dynamic provisioning of S3 buckets by Kubernetes users.
+
+### Overview
+
+  Rook-Ceph has developed Custom Resource Definitions to support the dynamic provisioning of Ceph Object Stores and Ceph Object Store Users.  This proposal introduces the next logical step of dynamic bucket provisioning with the addition of a CephObjectBucketClaim CRD.  The intent is to round out the Rook-Ceph experience for Kubernetes users by providing a single control plane for the managing of Ceph Object components.      
+
+## Goals
+
+- Provide Rook-Ceph users the ability to dynamically provision object buckets via the Kubernetes API with a CephObjectBucketClaim Custom Resource Definition (CRD).
+- Utilize a familiar and deterministic pattern when injecting bucket connection information into workload environments. 
+
+## Non-Goals
+
+- This proposal does not implement a generalized method for bucket provisioning; it will be specific to Rook-Ceph
+- This design does not attempt to implement a 1:1 model of Kubernetes PersistentVolumes and PersistentVolumeClaims.  Some shallow similarities will exist. 
+- This design does not include the Swift interface implementation by Ceph Object.
+- This design does not provide users a means of deleting buckets via the Kubernetes API.  This creates a dangerous situation which could lead to accidental loss of data.  Only the Kubernetes API objects are deleted.
+
+## Users
+
+- Admin: A Kubernetes cluster administrator with permissions to instantiate and control access to cluster storage. The admin will use existing Rook-Ceph CRDs to create object stores and StorageClasses to make them accessible to users.  
+- User: A Kubernetes cluster user with limited permissions, typically confined CRUD operations on Kubernetes objects within a single namespace.  The user will create request ceph object buckets by instantiating CephObjectBucketClaims. 
+
+## Use Cases
+
+### Assumptions
+
+1. A Rook-Ceph operator deployed on a Kubernetes Cluster.
+1. A CephObjectStore object exists along with the accompanying Service in a protected namespace.
+1. An existing CephObjectStoreUser exists along with the accompanying Secret in the user's namespace.
+
+##### Use Case: Expose an Object Store Endpoint for Bucket Provisioning
+
+As an admin, I want to expose an existing object store to cluster users so they can begin bucket provisioning via the Kubernetes API._
```

Comment:
Since the object store service is exposed today, what about adding "via a Storage Class" this use case?

Reply:
Maybe also add that these steps are unchanged from today other than creating the StoraeClass.